### PR TITLE
Cryptid CLI

### PR DESCRIPTION
**Pull request by whazzmaster**
This branch implements a CLI for the cryptid service inside the cryptid node module. It will allow users to login to the service (default is production URL, but override-able via the `CRYPTID_SERVER` env variable.

The cli will have a git-like subcommand architecture, such that the following commands will eventually be runnable:

```
$ cryptid login
$ cryptid logout
$ cryptid accounts list
$ cryptid accounts add --name "My New Account"
$ cryptid products list
$ cryptid products add --account-id 100 --name "My Product"
$ cryptid properties list
$ cryptid properties add --account-id 100 --product-id 101 --name "My Property"
$ cryptid users list --account-id 100
$ cryptid users add --account-id 100 --email jim@adorable.io 
```

### CODE REVIEW

File: src-bin/cryptid-accounts-create.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+let prefs = getPreferences();
+
+if (prefs.needsLogin) {
+  console.log(chalk.red('You must first login with "cryptid login"'));
+  process.exit(1);
+}
+
+if(!program.accountName) {
```
Comment (itsthatguy) [inline src-bin/cryptid-accounts-create.js]: could we get consistent code style?

File: package.json [hunk 2]
```diff
@@ -4,9 +4,15 @@
   "description": "javascript client to cryptid analytics",
   "website": "https://cryptid.adorable.io",
   "main": "dist/index.js",
+  "bin": {
+    "cryptid": "./bin/cryptid"
+  },
   "scripts": {
-    "build": "$(npm bin)/babel src/ --ignore src/__mocks__ -d dist --copy-files",
-    "watch": "$(npm bin)/babel src/ --ignore src/__mocks__ -d dist --copy-files --watch",
+    "build-lib": "$(npm bin)/babel src/ --ignore src/__mocks__ -d dist --copy-files",
```
Comment (itsthatguy) [inline package.json]: Prefer `build:lib` syntax

File: src-bin/cli.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    url: url,
+    method: 'post',
+    json: {
+      user: {
+        email: username,
+        password: password,
+      }
+    }
+  };
+  request(options, (error, response, body) => {
```
Comment (itsthatguy) [inline src-bin/cli.js]: remove unused parameters

File: src-bin/cli.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      }
+    }
+  };
+  request(options, callback);
+}
+
+export function updateAccount(accountId, accountName, callback) {
+  let url = buildUrl(`/api/accounts/${accountId}`);
+  let token = getPreferences().account.token;
+  let options = {
```
Comment (itsthatguy) [inline src-bin/cli.js]: given how much this options model is used, seems like we should have an options builder function.

**Comment by itsthatguy** [comment]
The files existing in `src-bin` seems odd to me. Typically, all source files live in `src`. Output distributed files to `dist`. Point the `bins` and `main` in `package.json` to the files found in the `dist` folder

File: src-bin/cryptid-accounts-create.js (hunk 1 above)
Comment (itsthatguy) [reply src-bin/cryptid-accounts-create.js]: `if (...)` vs `if(...)`

File: src/bin/cli.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+function buildUrl(path) {
+  return url.resolve(SERVER, path);
+}
+
+function loadSettings() {
+  let name = `com.adorable.cryptid.${md5(SERVER)}`;
+  let settings = new Preferences(name, {server: SERVER, token: '', email: ''});
+
+  settings.isLoggedIn = settings.token.length > 0 && settings.email.length > 0;
```
Comment (itsthatguy) [inline src/bin/cli.js:18]: is it ever possible that settings could end up undefined or null?

Comment (whazzmaster) [reply src/bin/cli.js:18]: No, if it doesn't exist then it uses the template on line 16 to create a new instance.

**Review by veezus — APPROVED** [review]
Pending further comments from @itsthatguy

**Pull request merged**